src/data/polynomial/reverse.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.erase_lead
+
+/-!
+# Reverse of a univariate polynomial
+
+The main definition is `reverse`.  Applying `reverse` to a polynomial `f : polynomial R` produces
+the polynomial with a reversed list of coefficients, equivalent to `X^f.nat_degree * f(1/X)`.
+
+The main result is that `reverse (f * g) = reverse (f) * reverse (g)`, provided the leading
+coefficients of `f` and `g` do not multiply to zero.
+-/
+
+namespace polynomial
+
+open polynomial finsupp finset
+open_locale classical
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+namespace rev
```

Comment:
Why is this in a namespace `rev`? I would omit this.

Reply:
I removed the namespace in my local file!  It will get uploaded once I push!